### ISSUE TITLE
get Ness's trained single-star Cannon from her first Cannon paper

### ISSUE THREAD

**Issue by davidwhogg** [adrn/ebak#8]
cc @mkness

**Comment by mkness** [comment]
I think you should use at least a 4-parameter model, one that includes alpha/Fe. Do you object? If not I will send you a dropbox link to a DR12 4-parameter model instead (Teff, logg, Fe/H, alpha/Fe).  It would also be interesting to use the 5 parameter model with log age as a label as well..

**Comment by mkness** [comment]
Link to this sent via email. This model is for DR12 and uses the DR12 spectra+training labels and my continuum normalisation in the paper. 
This model is quadratic, only for giants and has 4 labels: Teff,logg, [Fe/H] ,[alpha/Fe].
The columns are:
training data, training labels, list of labels, offsets used, coeffs, covariance matrix, scatters, chisquared, per-pixel chisquared.
- code below to generate a spectrum using this model. 
- input coeffs and offsets from the coeffs pickle file
- input labels being the label of the test star you want to generate a spectrum for.

def makespec(coeffs, offsets, labels):
    nlabels = len(labels)
    features_data = np.hstack((1, labels - offsets))
    newfeatures_data = np.array([np.outer(labels-offsets, labels-offsets)[np.triu_indices(nlabels)] ])
    features_data_final = np.hstack((features_data, newfeatures_data.flatten()))
    model_gen = np.dot(coeffs,features_data_final.T)
    return model_gen

**Comment by davidwhogg** [comment]
Hey @adrn should we check this in? I am assigning this to you.

**Comment by mkness** [comment]
Implemented proof of concept binary Cannon model:  DWH you will hate the way this is done  ! nevertheless was very easy to put in in the bad way I did it !: 
![binary_model](https://cloud.githubusercontent.com/assets/937118/15594724/9e04c574-23b8-11e6-8de3-6a75d5989493.png)
Currently uses a very small training set and inconsistent continuum normalisation between data and model.
Teff = 4888, logg = 3.3, [Fe/H] = -0.26, alpha/Fe = 0.01
Teff = 5036, logg = 3.7, Fe/H = -0.138, alpha/Fe = -0.10

**Comment by davidwhogg** [comment]
What do you get for the two velocities? Let's close this issue once you answer that!

**Comment by mkness** [comment]
I don't derive them in the fit - they are an input: I use -6/+3 pixels though for the shift of each respectively. I fixed the continuum normalisation and the model is far improved and now I fit for 9 labels - stellar params + a few elements. [Below - a bit noisy as am using too few training data for  9 labels]. Can we add a section where we report the parameters for the star? Does anyone care about these? ??
![9labels](https://cloud.githubusercontent.com/assets/937118/15634339/99b9621c-25c1-11e6-90b9-80864b3d769d.png)
['teff', 'logg', 'feh', 'C', 'N', 'O', 'Na', 'Mg', 'Al']
star1 = [4787.652, 2.515, -0.015, -0.006, -0.114, 0.306, -0.085, -0.024, -0.026]
star2 = [4790.588, 2.538, 0.002, -0.009, -0.082, 0.25, -0.063, -0.043, -0.001]

 Also, I wrote bad code that returns similar stars that are spectroscopic binaries and I have a list of these, some below. Do we want to derive velocities and params for these, are these interesting? Examples are:  2M21081720+4900002, 2M19430973+2357587, 2M04523015+3925462.fits
